### PR TITLE
Amqp redesign refactor

### PR DESCRIPTION
**Pull request by davilu**
## Checklist
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [ ] This pull-request is submitted against the `master` branch.

### CODE REVIEW

File: iothub/device/src/Transport/DeviceIdentity.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Devices.Shared;
+
+namespace Microsoft.Azure.Devices.Client.Transport
+{
+    /// <summary>
+    /// Base class of DeviceClientEndpointIdentity
```
Comment (zolvarga) [inline iothub/device/src/Transport/DeviceIdentity.cs]: >DeviceClientEndpointIdentity [](start = 22, length = 28)

Change the comment to new name

File: iothub/device/src/Transport/Amqp/AmqpConnectionPool.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+using Microsoft.Azure.Devices.Shared;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.Client.Transport.Amqp
+{
+   
+    class AmqpConnectionPool
```
Comment (zolvarga) [inline iothub/device/src/Transport/Amqp/AmqpConnectionPool.cs]: >AmqpConnectionPool [](start = 10, length = 18)

Create interface for testablity

File: iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs [hunk 3]
```diff
@@ -0,0 +1,136 @@
+﻿using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Devices.Client.Exceptions;
+using Microsoft.Azure.Devices.Shared;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.Client.Transport.Amqp
+{
+    internal class AmqpAuthenticationRefresher : IDisposable
```
Comment (zolvarga) [inline iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs]: >AmqpAuthenticationRefresher [](start = 19, length = 27)

Create interface for testablity

File: iothub/device/src/Transport/Amqp/AmqpLinkHelper.cs [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+using Microsoft.Azure.Amqp.Framing;
+using Microsoft.Azure.Devices.Shared;
+using System;
+using System.Globalization;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.Client.Transport.Amqp
+{
+    internal class AmqpLinkHelper
```
Comment (zolvarga) [inline iothub/device/src/Transport/Amqp/AmqpLinkHelper.cs]: >AmqpLinkHelper [](start = 19, length = 14)

Create interface for testablity

File: iothub/device/src/Transport/Amqp/AmqpUnitManager.cs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+using Microsoft.Azure.Devices.Shared;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.Client.Transport.Amqp
+{
+    class AmqpUnitManager
```
Comment (zolvarga) [inline iothub/device/src/Transport/Amqp/AmqpUnitManager.cs]: >AmqpUnitManager [](start = 10, length = 15)

Create interface for testablity

File: iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+    {
+        static readonly TimeSpan BufferPeriod = TimeSpan.FromSeconds(120);
+        private readonly AmqpCbsLink AmqpCbsLink;
+        private readonly IotHubConnectionString ConnectionString;
+        private readonly string Audience;
```
Comment (zolvarga) [inline iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs]: >Audience [](start = 32, length = 8)

Example for audience values. Devicename = z-001

"z-test.azure-devices.net/devices/z-001/messages/events"
"z-test.azure-devices.net/devices/z-001/messages/deviceBound"
"z-test.azure-devices.net/devices/z-001/methods/deviceBound"
"z-test.azure-devices.net/devices/z-001/methods/deviceBound"
"z-test.azure-devices.net/devices/z-001/twin"
"z-test.azure-devices.net/devices/z-001/twin"

File: iothub/device/src/Transport/Amqp/AmqpLinkHelper.cs (hunk 4 above)
Comment (davilu) [reply iothub/device/src/Transport/Amqp/AmqpLinkHelper.cs]: This class contains all static functions, not suitable for interface.

File: iothub/device/src/Transport/Amqp/AmqpConnectionPool.cs (hunk 2 above)
Comment (davilu) [reply iothub/device/src/Transport/Amqp/AmqpConnectionPool.cs]: done

File: iothub/device/src/Transport/Amqp/AmqpUnitManager.cs (hunk 5 above)
Comment (davilu) [reply iothub/device/src/Transport/Amqp/AmqpUnitManager.cs]: done

File: iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs (hunk 3 above)
Comment (davilu) [reply iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs]: done

File: iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs (hunk 6 above)
Comment (davilu) [reply iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs]: Go with `endpoint+device id`

**Comment by CIPop** [comment]
Moving to `dev` for now.

**Pull request closed without merging**